### PR TITLE
FTUE - Dashboard

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
FTUE part 1 - Personal Dashboard

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/47aa9bb4-6967-4a96-83c7-11bd6b20f1eb/7e38eab4-890a-4fc3-8f48-a1f9bd1ab822?c=3a4c80ae-7cd6-49eb-8257-ea0c11f63723

### CODE REVIEW

File: webapp/src/components/workspaceSwitcher/__snapshots__/workspaceSwitcher.test.tsx.snap [hunk 1]
```diff
@@ -129,7 +140,7 @@ exports[`components/workspaceSwitcher/WorkspaceSwitcher no more workspaces avail
       <span
         id="aria-context"
       >
-          Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
+         1 result available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
```
Comment (justinegeffen) [inline webapp/src/components/workspaceSwitcher/__snapshots__/workspaceSwitcher.test.tsx.snap:143]:
```suggestion
         1 result available. Use the Up and Down arrows on your keyboard to choose options. Press Enter to select the currently focused option. Press Escape to exit the menu.
```

File: webapp/src/components/workspaceSwitcher/__snapshots__/workspaceSwitcher.test.tsx.snap [hunk 2]
```diff
@@ -17,7 +17,7 @@ exports[`components/workspaceSwitcher/WorkspaceSwitcher 2 more workspaces availa
       <span
         id="aria-context"
       >
-         2 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
+         3 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
```
Comment (justinegeffen) [inline webapp/src/components/workspaceSwitcher/__snapshots__/workspaceSwitcher.test.tsx.snap:20]:
```suggestion
         3 results available. Use the Up and Down arrows on your keyboard to choose options. Press Enter to select the currently focused option. Press Escape to exit the menu.
```

File: webapp/src/components/workspaceSwitcher/__snapshots__/workspaceOptions.test.tsx.snap [hunk 3]
```diff
@@ -129,7 +140,7 @@ exports[`components/workspaceSwitcher/WorkspaceOptions no more workspaces availa
       <span
         id="aria-context"
       >
-          Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
+         1 result available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
```
Comment (justinegeffen) [inline webapp/src/components/workspaceSwitcher/__snapshots__/workspaceOptions.test.tsx.snap:143]:
```suggestion
         1 result available. Use the Up and Down arrows on your keyboard to choose options. Press Enter to select the currently focused option. Press Escape to exit the menu.
```

File: webapp/src/components/workspaceSwitcher/__snapshots__/workspaceOptions.test.tsx.snap [hunk 4]
```diff
@@ -17,7 +17,7 @@ exports[`components/workspaceSwitcher/WorkspaceOptions 2 more workspaces availab
       <span
         id="aria-context"
       >
-         2 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
+         3 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu.
```
Comment (justinegeffen) [inline webapp/src/components/workspaceSwitcher/__snapshots__/workspaceOptions.test.tsx.snap:20]:
```suggestion
         3 results available. Use the Up and Down arrows on your keyboard to choose options. Press Enter to select the currently focused option. Press Escape to exit the menu.
```

**Review by justinegeffen — COMMENTED** [review]
Some minor edits and a suggested rewrite for the results content. Happy to discuss that further!

**Review by justinegeffen — APPROVED** [review]

**Review by sbishel — CHANGES_REQUESTED** [review]
I ran into a few issues while testing - 

It matches figma, so more a question for @asaadmahmood, the button to hide the sidebar is no longer available.
Screen Shot 2021-09-13 at 10 41 39 AM

When in Dashboard, "Add Board" should not be available. (It actually adds to last viewed board, if you add a board)
Screen Shot 2021-09-13 at 8 50 29 AM

Direct channels Title does not display in either Dropdown or Dashboard. Group messages appear fine.
Screen Shot 2021-09-13 at 8 49 14 AM


Screen Shot 2021-09-13 at 8 49 30 AM

Screen Shot 2021-09-13 at 10 40 21 AM

**Comment by harshilsharma63** [comment]
@sbishel 

> When in Dashboard, "Add Board" should not be available. (It actually adds to last viewed board, if you add a board)

Fixed. It's now hidden in the dashboard view.

> Direct channels Title does not display in either Dropdown or Dashboard. Group messages appear fine

It was already fixed in a separate PR (#1179).
Screenshot 2021-09-14 at 12 30 48 PM
Screenshot 2021-09-14 at 12 31 07 PM

File: webapp/src/components/sidebar/sidebar.tsx [hunk 5]
```diff
@@ -95,23 +97,47 @@ const Sidebar = React.memo((props: Props) => {
...
+
+            {
+                props.isDashboard &&
+                (
+                    <React.Fragment>
+                        <WorkspaceSwitcher/>
+                        <div className='Sidebar__onboarding'>
+                            <DashboardOnboardingSvg/>
+                            <div>
+                                {'Use the switcher to easily change channels'}
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply webapp/src/components/sidebar/sidebar.tsx:114]: Done.

File: webapp/src/components/sidebar/sidebar.tsx [hunk 6]
```diff
@@ -95,23 +97,47 @@ const Sidebar = React.memo((props: Props) => {
...
-                                nextBoardId={board.id === props.activeBoardId ? nextBoardId : undefined}
-                            />
-                        )
-                    })
-                }
-            </div>
+
+            {
+                workspace && workspace.id !== '0' && !props.isDashboard &&
+                <WorkspaceSwitcher activeWorkspace={workspace}/>
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply webapp/src/components/sidebar/sidebar.tsx:103]: Combining that with `workspace && workspace.id !== '0'` was making it look messy, so didn't do it.

**Comment by justinegeffen** [comment]
Added the docs needed label as the docs will need to be updated, but it's not a blocker to merge.

**Review by sbishel — APPROVED** [review]
Approving and merging. I did create a bug related to this PR or https://github.com/mattermost/focalboard/pull/1179

https://github.com/mattermost/focalboard/issues/1205

**Pull request merged**